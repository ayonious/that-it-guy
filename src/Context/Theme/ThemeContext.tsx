import React from "react";
import createThemeContext from "./createThemeContext";
import {
  darkTheme,
  mapThemeNameToDetails,
  ThemeDetails,
  Themes,
} from "./ThemeEnums";

interface ContextProps {
  theme: ThemeDetails;
}

const themeReducer = (
  state: ContextProps,
  action: { type: string; theme: string }
) => {
  switch (action.type) {
    case "change_theme":
      localStorage.setItem("selectedTheme", action.theme);
      return {
        theme: mapThemeNameToDetails[action.theme as Themes],
      };
    default:
      return state;
  }
};

const changeTheme = (dispatch: any) => {
  return (theme: string) => {
    dispatch({ type: "change_theme", theme });
  };
};

export const { Context, Provider } = createThemeContext(
  themeReducer,
  { changeTheme },
  {
    theme: darkTheme,
  }
);
