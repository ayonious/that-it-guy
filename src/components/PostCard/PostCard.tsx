import React from "react";
import { Link } from "gatsby";
import { CardArticleWrapper, InfoDivWrapper, CardImageWrapper } from "./styles";
import { LinkDivWrapper } from "../../css/commonstyles";

import ThemeContext from "../../Context/ThemeContext";

import Image from "gatsby-image";
import TagsList from "../TagsList/TagsList";

interface PostCardData {
  post: {
    frontmatter: {
      title: string;
      date: string;
      slug: string;
      tags: string[];
      image: {
        childImageSharp: {
          fluid: any;
        };
      };
    };
    excerpt: string;
  };
  key: number;
}

const PostCard = ({ post }: PostCardData) => {
  const { title, date, slug, tags } = post.frontmatter;
  const img = post.frontmatter.image.childImageSharp.fluid;

  return (
    <ThemeContext.Consumer>
      {({ theme }) => (
        <CardArticleWrapper theme={theme}>
          <CardImageWrapper>
            <Image fluid={img} />
          </CardImageWrapper>
          <InfoDivWrapper theme={theme}>
            <div>
              <h2>{title}</h2>
              <h6>
                <span> {date} </span>
              </h6>
              <TagsList tags={tags} />
              <p>{post.excerpt}</p>
              <LinkDivWrapper theme={theme}>
                <Link to={`blogs/${slug}`}>Read...</Link>
              </LinkDivWrapper>
            </div>
          </InfoDivWrapper>
        </CardArticleWrapper>
      )}
    </ThemeContext.Consumer>
  );
};

export default PostCard;
