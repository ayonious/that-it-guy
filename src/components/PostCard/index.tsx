import Image from "gatsby-image";
import { throttle } from "lodash";
import React, {
  createRef,
  useContext,
  useEffect,
  useRef,
  useState,
} from "react";

import { Context as ThemeContext } from "../../Context/Theme/ThemeContext";
import SlidingComponent from "../SlidingComponent";
import StyledButton from "../StyledButton";
import TagsList from "../TagsList";
import {
  ButtonWrapper,
  CardArticleWrapper,
  CardImageWrapper,
  FlexRowWrapper,
  HeadlineAndDateWrapper,
  InfoDateWrapper,
  InfoDivWrapper,
  InfoExcerptWrapper,
  InfoHeadlineWrapper,
  MiniCardImageWrapper,
} from "./styles";

export interface PostCardData {
  post: {
    frontmatter: {
      title: string;
      date: string;
      slug: string;
      tags: string[];
      image: {
        childImageSharp: {
          fluid: any;
        };
      };
    };
    excerpt: string;
  };
  key: number;
}

const PostCard = (props: PostCardData) => {
  const {
    post: {
      frontmatter: {
        title,
        date,
        slug,
        tags,
        image: {
          childImageSharp: { fluid: img },
        },
      },
      excerpt,
    },
  } = props;

  const {
    state: { theme },
  } = useContext(ThemeContext);

  const inputRef = useRef(null);
  const [visible, changeVisible] = useState(false);

  const scrollListener = throttle(() => {
    if (!inputRef.current) {
      return;
    }
    const { clientHeight } = document.documentElement;
    const top = inputRef.current.getBoundingClientRect().top;
    const scrollPxFromBottom = clientHeight - top;

    if (scrollPxFromBottom > 20) {
      changeVisible(true);
      document.removeEventListener("scroll", scrollListener, true);
    }
  }, 500);

  useEffect(() => {
    scrollListener(); // initially without scroll even make some articles visible
    document.addEventListener("scroll", scrollListener, true);
    return () => document.removeEventListener("scroll", scrollListener, true);
  });

  return (
    <SlidingComponent>
      <CardArticleWrapper theme={theme}>
        <CardImageWrapper>
          <Image fluid={img} />
        </CardImageWrapper>
        <InfoDivWrapper theme={theme}>
          <FlexRowWrapper>
            <MiniCardImageWrapper>
              <Image fluid={img} />
            </MiniCardImageWrapper>
            <HeadlineAndDateWrapper>
              <InfoHeadlineWrapper>{title}</InfoHeadlineWrapper>
              <InfoDateWrapper>{date}</InfoDateWrapper>
            </HeadlineAndDateWrapper>
          </FlexRowWrapper>
          <TagsList tags={tags} />
          <InfoExcerptWrapper>{excerpt}</InfoExcerptWrapper>
          <ButtonWrapper>
            <StyledButton
              to={`blogs/${slug}`}
              theme={theme}
              title={"Read More..."}
            />
          </ButtonWrapper>
        </InfoDivWrapper>
      </CardArticleWrapper>
    </SlidingComponent>
  );
};

export default PostCard;
