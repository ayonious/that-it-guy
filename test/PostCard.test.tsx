/**
 * @jest-environment jsdom
 */

import { render } from "@testing-library/react";
import { expect } from "@jest/globals";
import * as React from "react";

import PostCard from "../src/components/PostCard";
import { Provider as ThemeProvider } from "../src/Context/Theme/ThemeContext";

it("PostCard renders", () => {
  const postCard = {
    post: {
      frontmatter: {
        title: "React Hooks - memo",
        slug: "react-hooks-memo",
        date: "July 3rd, 2021",
        tags: ["js", "react", "hooks"],
        image: {
          childImageSharp: {
            gatsbyImageData: {
              layout: "fullWidth" as any,
              backgroundColor: "#b8a898",
              images: {
                fallback: {
                  src: "/static/902341f7675c60ef9661b2a4aa2de276/cb908/pexels-cottonbro-3832028.jpg",
                  srcSet:
                    "/static/902341f7675c60ef9661b2a4aa2de276/cb908/pexels-cottonbro-3832028.jpg 640w",
                  sizes: "100vw",
                },
                sources: [
                  {
                    srcSet:
                      "/static/902341f7675c60ef9661b2a4aa2de276/bd763/pexels-cottonbro-3832028.webp 640w",
                    type: "image/webp",
                    sizes: "100vw",
                  },
                ],
              },
              width: 1,
              height: 1.5,
            },
          },
        },
      },
      excerpt:
        "In simple words this controls when to not re-render your component. I got a situation recently where a Form component was re-rendering andâ€¦",
    },
    blogUrl: "blogs/react-hooks-memo",
  };

  const tree = render(
    <ThemeProvider>
      <PostCard post={postCard.post} key={1} blogUrl="/" />
    </ThemeProvider>
  );
  expect(tree).toMatchSnapshot();
});
