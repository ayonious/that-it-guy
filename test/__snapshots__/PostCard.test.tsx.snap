// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostCard renders 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnxRM hibReb"
      >
        <article
          class="sc-eCApnc fxjQkC"
        >
          <div
            class="sc-jSFjdj jnRzUH"
          >
            <div
              class=" gatsby-image-wrapper"
              style="position: relative; overflow: hidden;"
            >
              <div
                aria-hidden="true"
                style="width: 100%; padding-bottom: 56.5%;"
              />
              <img
                alt=""
                aria-hidden="true"
                src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAALABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAMEAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAABgbGxOTCv/8QAGRAAAwADAAAAAAAAAAAAAAAAAQIQAxES/9oACAEBAAEFAkCk5E5M3P/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQIBAT8BP//EABYQAAMAAAAAAAAAAAAAAAAAAAERIP/aAAgBAQAGPwIOv//EABoQAQADAAMAAAAAAAAAAAAAAAEAEBEhMZH/2gAIAQEAAT8hMnhpDwPInNKe2v/aAAwDAQACAAMAAAAQzA//xAAWEQEBAQAAAAAAAAAAAAAAAAARECH/2gAIAQMBAT8QcJ//xAAWEQADAAAAAAAAAAAAAAAAAAAQESH/2gAIAQIBAT8QVH//xAAaEAEBAQEAAwAAAAAAAAAAAAABEQBBMVFx/9oACAEBAAE/ELF4lvNCAeV6fmnEMOeqr7XXf//Z"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
              />
              <picture>
                <source
                  sizes="(max-width: 640px) 100vw, 640px"
                  srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/61e93/person-holding-compass-841286.webp 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/1f5c5/person-holding-compass-841286.webp 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/991d2/person-holding-compass-841286.webp 640w"
                  type="image/webp"
                />
                <source
                  sizes="(max-width: 640px) 100vw, 640px"
                  srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w"
                />
                <img
                  alt=""
                  loading="lazy"
                  sizes="(max-width: 640px) 100vw, 640px"
                  src="/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg"
                  srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
                />
              </picture>
              <noscript>
                &lt;picture&gt;&lt;source type='image/webp' srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/61e93/person-holding-compass-841286.webp 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/1f5c5/person-holding-compass-841286.webp 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/991d2/person-holding-compass-841286.webp 640w" sizes="(max-width: 640px) 100vw, 640px" /&gt;&lt;source srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w" sizes="(max-width: 640px) 100vw, 640px" /&gt;&lt;img loading="lazy" sizes="(max-width: 640px) 100vw, 640px" srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w" src="/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
              </noscript>
            </div>
          </div>
          <div
            class="sc-pNWdM kaQAYE"
          >
            <div
              class="sc-iCoGMd fHAOGr"
            >
              <div
                class="sc-gKAaRy cLxduy"
              >
                <div
                  class=" gatsby-image-wrapper"
                  style="position: relative; overflow: hidden;"
                >
                  <div
                    aria-hidden="true"
                    style="width: 100%; padding-bottom: 56.5%;"
                  />
                  <img
                    alt=""
                    aria-hidden="true"
                    src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAALABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAMEAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAABgbGxOTCv/8QAGRAAAwADAAAAAAAAAAAAAAAAAQIQAxES/9oACAEBAAEFAkCk5E5M3P/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQIBAT8BP//EABYQAAMAAAAAAAAAAAAAAAAAAAERIP/aAAgBAQAGPwIOv//EABoQAQADAAMAAAAAAAAAAAAAAAEAEBEhMZH/2gAIAQEAAT8hMnhpDwPInNKe2v/aAAwDAQACAAMAAAAQzA//xAAWEQEBAQAAAAAAAAAAAAAAAAARECH/2gAIAQMBAT8QcJ//xAAWEQADAAAAAAAAAAAAAAAAAAAQESH/2gAIAQIBAT8QVH//xAAaEAEBAQEAAwAAAAAAAAAAAAABEQBBMVFx/9oACAEBAAE/ELF4lvNCAeV6fmnEMOeqr7XXf//Z"
                    style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
                  />
                  <picture>
                    <source
                      sizes="(max-width: 640px) 100vw, 640px"
                      srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/61e93/person-holding-compass-841286.webp 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/1f5c5/person-holding-compass-841286.webp 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/991d2/person-holding-compass-841286.webp 640w"
                      type="image/webp"
                    />
                    <source
                      sizes="(max-width: 640px) 100vw, 640px"
                      srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w"
                    />
                    <img
                      alt=""
                      loading="lazy"
                      sizes="(max-width: 640px) 100vw, 640px"
                      src="/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg"
                      srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w"
                      style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
                    />
                  </picture>
                  <noscript>
                    &lt;picture&gt;&lt;source type='image/webp' srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/61e93/person-holding-compass-841286.webp 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/1f5c5/person-holding-compass-841286.webp 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/991d2/person-holding-compass-841286.webp 640w" sizes="(max-width: 640px) 100vw, 640px" /&gt;&lt;source srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w" sizes="(max-width: 640px) 100vw, 640px" /&gt;&lt;img loading="lazy" sizes="(max-width: 640px) 100vw, 640px" srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w" src="/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
                  </noscript>
                </div>
              </div>
              <div
                class="sc-fujyAs hlijaf"
              >
                <span
                  class="sc-jrsJWt lhrjyw"
                >
                  React Reducer a Short Example
                </span>
                <div
                  class="sc-kEqXSa fKhgzJ"
                >
                  May 29th, 2020
                </div>
              </div>
            </div>
            <div
              class="sc-dlnjwi gpAwct"
            >
              <div
                class="sc-hKFxyN kTJRAN"
              >
                #
                js
              </div>
              <div
                class="sc-hKFxyN kTJRAN"
              >
                #
                react
              </div>
              <div
                class="sc-hKFxyN kTJRAN"
              >
                #
                reducer
              </div>
            </div>
            <p
              class="sc-iqAclL iFznqS"
            >
              I often see this thing but never understood how it could be used for making life easier. Finally found some good examples and decided to…
            </p>
            <div
              class="sc-crzoAE fwYfox"
            >
              <div
                class="sc-gtsrHT gXEQGZ"
              >
                <a
                  aria-current="page"
                  class=""
                  href="/"
                >
                  Read More...
                </a>
              </div>
            </div>
          </div>
        </article>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnxRM hibReb"
    >
      <article
        class="sc-eCApnc fxjQkC"
      >
        <div
          class="sc-jSFjdj jnRzUH"
        >
          <div
            class=" gatsby-image-wrapper"
            style="position: relative; overflow: hidden;"
          >
            <div
              aria-hidden="true"
              style="width: 100%; padding-bottom: 56.5%;"
            />
            <img
              alt=""
              aria-hidden="true"
              src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAALABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAMEAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAABgbGxOTCv/8QAGRAAAwADAAAAAAAAAAAAAAAAAQIQAxES/9oACAEBAAEFAkCk5E5M3P/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQIBAT8BP//EABYQAAMAAAAAAAAAAAAAAAAAAAERIP/aAAgBAQAGPwIOv//EABoQAQADAAMAAAAAAAAAAAAAAAEAEBEhMZH/2gAIAQEAAT8hMnhpDwPInNKe2v/aAAwDAQACAAMAAAAQzA//xAAWEQEBAQAAAAAAAAAAAAAAAAARECH/2gAIAQMBAT8QcJ//xAAWEQADAAAAAAAAAAAAAAAAAAAQESH/2gAIAQIBAT8QVH//xAAaEAEBAQEAAwAAAAAAAAAAAAABEQBBMVFx/9oACAEBAAE/ELF4lvNCAeV6fmnEMOeqr7XXf//Z"
              style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
            />
            <picture>
              <source
                sizes="(max-width: 640px) 100vw, 640px"
                srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/61e93/person-holding-compass-841286.webp 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/1f5c5/person-holding-compass-841286.webp 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/991d2/person-holding-compass-841286.webp 640w"
                type="image/webp"
              />
              <source
                sizes="(max-width: 640px) 100vw, 640px"
                srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w"
              />
              <img
                alt=""
                loading="lazy"
                sizes="(max-width: 640px) 100vw, 640px"
                src="/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg"
                srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w"
                style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
              />
            </picture>
            <noscript>
              &lt;picture&gt;&lt;source type='image/webp' srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/61e93/person-holding-compass-841286.webp 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/1f5c5/person-holding-compass-841286.webp 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/991d2/person-holding-compass-841286.webp 640w" sizes="(max-width: 640px) 100vw, 640px" /&gt;&lt;source srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w" sizes="(max-width: 640px) 100vw, 640px" /&gt;&lt;img loading="lazy" sizes="(max-width: 640px) 100vw, 640px" srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w" src="/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
            </noscript>
          </div>
        </div>
        <div
          class="sc-pNWdM kaQAYE"
        >
          <div
            class="sc-iCoGMd fHAOGr"
          >
            <div
              class="sc-gKAaRy cLxduy"
            >
              <div
                class=" gatsby-image-wrapper"
                style="position: relative; overflow: hidden;"
              >
                <div
                  aria-hidden="true"
                  style="width: 100%; padding-bottom: 56.5%;"
                />
                <img
                  alt=""
                  aria-hidden="true"
                  src="data:image/jpeg;base64,/9j/2wBDABALDA4MChAODQ4SERATGCgaGBYWGDEjJR0oOjM9PDkzODdASFxOQERXRTc4UG1RV19iZ2hnPk1xeXBkeFxlZ2P/2wBDARESEhgVGC8aGi9jQjhCY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2NjY2P/wgARCAALABQDASIAAhEBAxEB/8QAGAAAAwEBAAAAAAAAAAAAAAAAAAMEAQL/xAAVAQEBAAAAAAAAAAAAAAAAAAAAAf/aAAwDAQACEAMQAAABgbGxOTCv/8QAGRAAAwADAAAAAAAAAAAAAAAAAQIQAxES/9oACAEBAAEFAkCk5E5M3P/EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQMBAT8BP//EABQRAQAAAAAAAAAAAAAAAAAAABD/2gAIAQIBAT8BP//EABYQAAMAAAAAAAAAAAAAAAAAAAERIP/aAAgBAQAGPwIOv//EABoQAQADAAMAAAAAAAAAAAAAAAEAEBEhMZH/2gAIAQEAAT8hMnhpDwPInNKe2v/aAAwDAQACAAMAAAAQzA//xAAWEQEBAQAAAAAAAAAAAAAAAAARECH/2gAIAQMBAT8QcJ//xAAWEQADAAAAAAAAAAAAAAAAAAAQESH/2gAIAQIBAT8QVH//xAAaEAEBAQEAAwAAAAAAAAAAAAABEQBBMVFx/9oACAEBAAE/ELF4lvNCAeV6fmnEMOeqr7XXf//Z"
                  style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 1; transition-delay: 500ms;"
                />
                <picture>
                  <source
                    sizes="(max-width: 640px) 100vw, 640px"
                    srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/61e93/person-holding-compass-841286.webp 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/1f5c5/person-holding-compass-841286.webp 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/991d2/person-holding-compass-841286.webp 640w"
                    type="image/webp"
                  />
                  <source
                    sizes="(max-width: 640px) 100vw, 640px"
                    srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w"
                  />
                  <img
                    alt=""
                    loading="lazy"
                    sizes="(max-width: 640px) 100vw, 640px"
                    src="/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg"
                    srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w"
                    style="position: absolute; top: 0px; left: 0px; width: 100%; height: 100%; object-fit: cover; object-position: center; opacity: 0; transition: opacity 500ms;"
                  />
                </picture>
                <noscript>
                  &lt;picture&gt;&lt;source type='image/webp' srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/61e93/person-holding-compass-841286.webp 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/1f5c5/person-holding-compass-841286.webp 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/991d2/person-holding-compass-841286.webp 640w" sizes="(max-width: 640px) 100vw, 640px" /&gt;&lt;source srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w" sizes="(max-width: 640px) 100vw, 640px" /&gt;&lt;img loading="lazy" sizes="(max-width: 640px) 100vw, 640px" srcset="/static/b8db2eac530443fabe2a1827b2fb9c52/f836f/person-holding-compass-841286.jpg 200w,
/static/b8db2eac530443fabe2a1827b2fb9c52/2244e/person-holding-compass-841286.jpg 400w,
/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg 640w" src="/static/b8db2eac530443fabe2a1827b2fb9c52/f422e/person-holding-compass-841286.jpg" alt="" style="position:absolute;top:0;left:0;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center"/&gt;&lt;/picture&gt;
                </noscript>
              </div>
            </div>
            <div
              class="sc-fujyAs hlijaf"
            >
              <span
                class="sc-jrsJWt lhrjyw"
              >
                React Reducer a Short Example
              </span>
              <div
                class="sc-kEqXSa fKhgzJ"
              >
                May 29th, 2020
              </div>
            </div>
          </div>
          <div
            class="sc-dlnjwi gpAwct"
          >
            <div
              class="sc-hKFxyN kTJRAN"
            >
              #
              js
            </div>
            <div
              class="sc-hKFxyN kTJRAN"
            >
              #
              react
            </div>
            <div
              class="sc-hKFxyN kTJRAN"
            >
              #
              reducer
            </div>
          </div>
          <p
            class="sc-iqAclL iFznqS"
          >
            I often see this thing but never understood how it could be used for making life easier. Finally found some good examples and decided to…
          </p>
          <div
            class="sc-crzoAE fwYfox"
          >
            <div
              class="sc-gtsrHT gXEQGZ"
            >
              <a
                aria-current="page"
                class=""
                href="/"
              >
                Read More...
              </a>
            </div>
          </div>
        </div>
      </article>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
